From a1805d83ec0258bc90e49566f204db9d6a903a49 Mon Sep 17 00:00:00 2001
From: Arjan van de Ven <arjan@linux.intel.com>
Date: Thu, 20 Aug 2015 20:26:53 +0000
Subject: [PATCH 08/12] cpu: reduce the damage from intel_pt by bailing out on
 cpus that don't support it

the intel_pt code does a series of expensive things
BEFORE it changes the cpu capabilities (and then bail out).

This is silly, bail out immediately on CPUs that don't have
support for the feature.

This changes reduces out boot time by over 10%

Signed-off-by: Arjan van de Ven <arjan@linux.intel.com>
Signed-off-by: Miguel Bernal Marin <miguel.bernal.marin@linux.intel.com>
---
 arch/x86/kernel/cpu/perf_event_intel_pt.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/x86/kernel/cpu/perf_event_intel_pt.c b/arch/x86/kernel/cpu/perf_event_intel_pt.c
index 4216928..f583013 100644
--- a/arch/x86/kernel/cpu/perf_event_intel_pt.c
+++ b/arch/x86/kernel/cpu/perf_event_intel_pt.c
@@ -1129,6 +1129,10 @@ static __init int pt_init(void)
 {
 	int ret, cpu, prior_warn = 0;
 
+
+	if (!test_cpu_cap(&boot_cpu_data, X86_FEATURE_INTEL_PT))
+		return -ENODEV;
+
 	BUILD_BUG_ON(sizeof(struct topa) > PAGE_SIZE);
 	get_online_cpus();
 	for_each_online_cpu(cpu) {
-- 
2.4.3

